src/test/java/picard/analysis/CollectOxoGMetricsTest.java

```diff
@@ -0,0 +1,129 @@
+package picard.analysis;
+
+import htsjdk.samtools.metrics.MetricsFile;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.HashSet;
+
+public class CollectOxoGMetricsTest {
+    private static final File TEST_DATA_DIR = new File("testdata/picard/sam/");
+    private static final File SAM_FILE = new File(TEST_DATA_DIR, "summary_alignment_stats_test.sam");
+    private static final File REFERENCE_SEQUENCE = new File(TEST_DATA_DIR, "merger.fasta");
+
+    @Test
+    public void testCollectOxoGMetrics() throws IOException {
+        final File outputFile = File.createTempFile("test", ".oxo_g_metrics", TEST_DATA_DIR);
+        outputFile.deleteOnExit();
+        final String[] args = new String[]{
+                "INPUT=" + SAM_FILE.getAbsolutePath(),
+                "OUTPUT=" + outputFile.getAbsolutePath(),
+                "REFERENCE_SEQUENCE=" + REFERENCE_SEQUENCE.getAbsolutePath()
+        };
+        CollectOxoGMetrics collectOxoGMetrics = new CollectOxoGMetrics();
+        Assert.assertEquals(collectOxoGMetrics.instanceMain(args), 0,
+                "Can't process " + SAM_FILE.getAbsolutePath() + " correctly");
+
+        final MetricsFile<CollectOxoGMetrics.CpcgMetrics, Comparable<?>> output = new MetricsFile<>();
+        output.read(new FileReader(outputFile));
+
+        final CollectOxoGMetrics.CpcgMetrics cpcgMetrics = output.getMetrics().get(4);
```

Comment:
It's not clear what index 4 refers to, could you create a variable with a useful name?